pkg/controller/import-controller.go

```diff
@@ -771,31 +843,160 @@ func createImportPodNameFromPvc(pvc *corev1.PersistentVolumeClaim) string {
 // name, and pvc. A nil secret means the endpoint credentials are not passed to the
 // importer pod.
-func createImporterPod(log logr.Logger, client client.Client, image, verbose, pullPolicy string, podEnvVar *importPodEnvVar, pvc *corev1.PersistentVolumeClaim, scratchPvcName *string, vddkImageName *string, priorityClassName string, installerLabels map[string]string) (*corev1.Pod, error) {
-	podResourceRequirements, err := GetDefaultPodResourceRequirements(client)
+func createImporterPod(log logr.Logger, client client.Client, args *importerPodArgs, installerLabels map[string]string) (*corev1.Pod, error) {
+	var err error
+	args.podResourceRequirements, err = GetDefaultPodResourceRequirements(client)
 	if err != nil {
 		return nil, err
 	}
 
-	workloadNodePlacement, err := GetWorkloadNodePlacement(client)
+	args.workloadNodePlacement, err = GetWorkloadNodePlacement(client)
 	if err != nil {
 		return nil, err
 	}
 
-	pod := makeImporterPodSpec(pvc.Namespace, image, verbose, pullPolicy, podEnvVar, pvc, scratchPvcName, podResourceRequirements, workloadNodePlacement, vddkImageName, priorityClassName)
+	var pod *corev1.Pod
+	if getSource(args.pvc) == SourceRegistry && args.pvc.Annotations[AnnRegistryImportMethod] == string(cdiv1.RegistryPullNode) {
+		args.importImage, err = getRegistryImportImage(args.pvc)
+		if err != nil {
+			return nil, err
+		}
+		pod = makeNodeImporterPodSpec(args)
+	} else {
+		pod = makeImporterPodSpec(args)
+	}
+
 	util.SetRecommendedLabels(pod, installerLabels, "cdi-controller")
 
-	if err := client.Create(context.TODO(), pod); err != nil {
+	if err = client.Create(context.TODO(), pod); err != nil {
 		return nil, err
 	}
-	log.V(3).Info("importer pod created\n", "pod.Name", pod.Name, "pod.Namespace", pod.Namespace, "image name", image)
+
+	log.V(3).Info("importer pod created\n", "pod.Name", pod.Name, "pod.Namespace", pod.Namespace, "image name", args.image)
 	return pod, nil
 }
 
+// makeNodeImporterPodSpec creates and returns the node docker cache based importer pod spec based on the passed-in importImage and pvc.
+func makeNodeImporterPodSpec(args *importerPodArgs) *corev1.Pod {
+	// importer pod name contains the pvc name
+	podName, _ := args.pvc.Annotations[AnnImportPod]
+
+	volumes := []corev1.Volume{
+		{
+			Name: "shared-volume",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: DataVolName,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: args.pvc.Name,
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
+
+	importerContainer := makeImporterContainerSpec(args.image, args.verbose, args.pullPolicy)
+
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: args.pvc.Namespace,
+			Annotations: map[string]string{
+				AnnCreatedBy: "yes",
```

Comment:
Don't see `alpha.image.policy.openshift.io/resolve-names: '*'` added anywhere if imagestream

Reply:
This is not needed when you set `Imagestream.spec.lookupPolicy.local: true`. I initially suspected this was the reason for the  OwnerReference issue, but adding it changed nothing.

Reply:
Okay but is it reasonable to expect that to be set?  Will adding the annotations harm anything?